app/controllers/api/v1x0/portfolios_controller.rb

```diff
@@ -44,8 +44,22 @@ def show
 
       def destroy
         portfolio = Portfolio.find(params.require(:id))
-        if portfolio.discard
-          head :no_content
+        svc = Catalog::SoftDelete.new(portfolio)
+        key = svc.process.restore_key
+
+        if portfolio.discarded?
+          render :json => { :restore_key => key }
+        else
+          render :json => { :errors => portfolio.errors }, :status => :unprocessable_entity
```

Comment:
Is there a way we can just throw a custom error during the `SoftDelete` processing if something goes wrong and then add another `rescue_from` in the `ExceptionHandler` class? It would remove the if/else condition here (and the same thing for `SoftDeleteRestore`

Reply:
Sure, I modified the callback that this if/else condition was needed for to raise a `Discard::DiscardError` and rescue from that instead.